components/script/dom/htmlimageelement.rs

```diff
@@ -517,6 +530,10 @@ impl HTMLImageElement {
 
     /// <https://html.spec.whatwg.org/multipage/#abort-the-image-request>
     fn abort_request(&self, state: State, request: ImageRequestPhase) {
+        if matches!(state, State::Broken | State::Unavailable) {
```

Comment:
I don't think it should include `State::Unavailable`, see "This [img](https://html.spec.whatwg.org/multipage/embedded-content.html#the-img-element) element's [current request](https://html.spec.whatwg.org/multipage/images.html#current-request)'s [state](https://html.spec.whatwg.org/multipage/images.html#img-req-state) becomes [broken](https://html.spec.whatwg.org/multipage/images.html#img-error)"

Reply:
Also, like in the other place, let's reject at the end of the function, so that the handler sees the updates.